src/act_cell.py

```diff
@@ -96,10 +105,17 @@ def act_step(self,batch_mask,prob_compare,prob,counter,state,input,acc_outputs,a
 
         input_with_flags = tf.concat([binary_flag, input], 1)
 
+        if self.state_is_tuple:
+            (c, h) = tf.split(state, 2, 1)
+            state = tf.contrib.rnn.LSTMStateTuple(c, h)
+
         output, new_state = static_rnn(cell=self.cell, inputs=[input_with_flags], initial_state=state, scope=type(self.cell).__name__)
 
+        if self.state_is_tuple:
+            new_state = tf.concat(new_state, 1)
+
         with tf.variable_scope('sigmoid_activation_for_pondering'):
-            p = tf.squeeze(tf.layers.dense(new_state, 1, activation=tf.sigmoid))
+            p = tf.squeeze(tf.sigmoid(core_rnn_cell_impl._linear(new_state, 1, True)), squeeze_dims=1)
```

Comment:
I'd prefer to keep the `tf.layers.dense`, as it's probably best not to use internal tf implementations of functions. The squeeze bit is fine though.